usr/lib/ep11_stdll/ep11_specific.c

```diff
@@ -1259,6 +1260,38 @@ CK_BBOOL ep11tok_pkey_usage_ok(STDLL_TokData_t *tokdata, SESSION *session,
     return success;
 }
 
+CK_BBOOL ep11tok_pkey_check_attrs(CK_ATTRIBUTE_PTR attrs,
+                               CK_ULONG attrs_len, CK_ULONG type)
+{
+    CK_BBOOL val;
+    CK_RV rc = CKR_OK;
+
+    rc = get_bool_attribute_by_type(attrs, attrs_len, type, &val);
+    if (rc != CKR_OK) {
+        /*  Attributes may not be specified in the attrs,
+         *  returning default TRUE for CKA_EXTRACTABLE
+         *  and FALSE for all other attributes  */
+        return type == CKA_EXTRACTABLE ? TRUE: FALSE;
+    }
+    return val;
+}
+
+CK_RV ep11tok_pkey_check_aes_xts(STDLL_TokData_t *tokdata, CK_ATTRIBUTE_PTR attrs,
+                                 CK_ULONG attrs_len)
+{
+    if (ep11tok_is_mechanism_supported(tokdata, CKM_AES_XTS_KEY_GEN) != CKR_OK) {
+        TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_INVALID));
+        return CKR_MECHANISM_INVALID;
+    }
+
+    if (ep11tok_pkey_check_attrs(attrs, attrs_len, CKA_EXTRACTABLE) ||
```

Comment:
Is there a reason why function `ep11tok_pkey_check_aes_xts()` gets the attribute array and number of attributes, and not simply the key object (`OBJECT *key_obj`)? At both places where `ep11tok_pkey_check_aes_xts()` is called (key gen and import), the to-be-created object is already available (`object_mgr_create_skel()` has already been done). 

If `ep11tok_pkey_check_aes_xts()` would get the OBJECT, you would not need function `ep11tok_pkey_check_attrs()`  at all, but simply could use the following (like it is done in `ep11tok_pkey_check()` ):

```
        if (object_is_extractable(key_obj) ||
            !object_is_pkey_extractable(key_obj) ||
            object_is_attr_bound(key_obj))
....
```

This will result to the same behavior as today, but its a little less code, and even more important, the code is consistent to the already existing pkey related code.